app/models/pglogical_subscription.rb

```diff
@@ -78,10 +78,12 @@ def self.delete_all(list = nil)
   end
 
   def disable
+    return unless logical_replication_supported?
```

Comment:
all of these `return unless logical_replication_supported?` are laborious.

Would it make sense to create a subclass of `PGLogicalSubscription` that just returns from all of these calls?
(a subclass is just an alternative for an `if`/`case` statement)

It would require changing the 3 calls to `new`, so I'm not sure if this is a valid suggestion.

Reply:
Oh like a null object pattern on `PGLogicalSubscription`?  That's actually a nice idea.

Reply:
Agreed.  The problem was that I didn't want to lose the context in each method as some of them needed to return `[]` or `nil` or `false` based on the context as I really wanted to ensure calling code didn't try to do anything with replication based on result.  For example, `subscriptions` needs to return an empty array as most callers assume they can loop on the result.

Reply:
Well, we would create the class only once and it would be use across all calls.
The factory is just going to return one of 2 classes based upon the super user status

Reply:
This is fixed, see comment in the new code.